Assets/Scripts/Characters/EnemyMVC/EnemyView.cs

```diff
@@ -0,0 +1,219 @@
+
+using System;
+using System.Collections;
+using Tanks.tank;
+using UnityEngine;
+using UnityEngine.AI;
+
+public class EnemyView : MonoBehaviour,IDamegable
+{
+    public EnemyController _EnemyController { get; private set; }
+    public NavMeshAgent agent;
+    public float range; //radius of sphere
+    public ParticleSystem EnemyExplosion;
+    [SerializeField]
+    private float PSDelay;
+    private float playerSqrDistance;
+    [SerializeField]
+    private float chaseSqrRadius = 900f;
+    [SerializeField]
+    private float attackSqrRadius = 100f;
+    public Transform centrePoint; //centre of the area the agent wants to move around in
+                                  //instead of centrePoint you can set it as the transform of the agent if you don't care about a specific area
+
+    public bool EnemyisMoving { get; private set; }
+    [SerializeField]
+    private TankView player;
+    [SerializeField]
+    private BulletServices bulletServices;
+
+    [SerializeField]
+    private EnemyStates currentState;
+    private EnemyIdleState idleState;
+    [SerializeField]
+    private EnemyPatrolling patrolState;
+    [SerializeField]
+    private EnemyAttackState attackState;
+    [SerializeField]
+    private EnemyChaseState chaseState;
+    
+    public Transform shootPoint;
+
+    [SerializeField]
+    private float speed;
+    private int health;
+    [SerializeField]
+    private UIManager uIManager;
+    public EnemyServices enemyServices { get; private set; }
+    public float ShootDelay { get; private set; }
+
+    private float timer;
+
+    void Start()
+    {
+        Init();
+    }
+    private void Init()
+    {
+        patrolState = GetComponent<EnemyPatrolling>();
+        attackState = GetComponent<EnemyAttackState>();
+        chaseState = GetComponent<EnemyChaseState>();
+        EnemyisMoving = true;
+        currentState = patrolState;
+        currentState.OnEnterState();
+        agent = GetComponent<NavMeshAgent>();
+        player = GameObject.FindGameObjectWithTag("Player").GetComponent<TankView>();
```

Comment:
You must optimize line 65 at all costs. Create a field in EnemyServices to hold the reference of player , when you instantiate the player , give the reference to EnemyService and then use EnemyServices to get the player ref to EnemyView.